projects/aws/bottlerocket-bootstrap/pkg/kubeadm/controlplane_join.go

```diff
@@ -115,9 +121,20 @@ func controlPlaneJoin() error {
 		return err
 	}
 
+	k8s131Compare, err := kubeadmVersion.Compare("1.31.0")
+	if err != nil {
+		return errors.Wrap(err, "Error comparing kubeadm version with v1.31.0")
+	}
+
 	// finish kubeadm
-	cmd = exec.Command(kubeadmBinary, utils.LogVerbosity, "join", "--skip-phases", "preflight,control-plane-prepare,kubelet-start,control-plane-join/etcd",
-		"--config", kubeadmJoinFile)
+	// 'kubelet-wait-bootstrap' phase was introduced only in 1.31, skip the phase for 1.31 and above
+	if k8s131Compare != -1 {
+		cmd = exec.Command(kubeadmBinary, utils.LogVerbosity, "join", "--skip-phases", "preflight,control-plane-prepare,kubelet-start,control-plane-join/etcd,kubelet-wait-bootstrap",
```

Comment:
Instead of the two exec calls, id probably define the string of skipped phases in the if and then just have the exec code once. same for below too